bluesky_widgets/qt/search_results.py

```diff
@@ -136,7 +123,7 @@ def _fetch_data(self, index):
             return self._data[index]
         else:
             self._data[index] = LOADING_PLACEHOLDER
-            self._request_queue.put(index)
+            self._data_loader.request(index)
```

Comment:
I don't think _fetch_data is necessary. You could instead in data() in line 174/187, directly emit a list of QModelIndex (or QModelIndexList) to DataLoader.request(). Then when data arrives the DataLoader will then do what it is doing now which is emitting back to self.dataChanged index and value. Also, removing this function will remove self._data being used in two places and keep the two objects separated.

Reply:
I'm sold. Thanks for pointing this out.